typespec-extension/src/code-model-builder.ts

```diff
@@ -1123,7 +1123,7 @@ export class CodeModelBuilder {
         },
         language: {
           default: {
-            serializedName: param.name,
+            serializedName: this.getSerializedName(param.param),
```

Comment:
After TCGC, we should not use this.

Reply:
I will try to rewrite this part to use TCGC's `(SdkQueryParameter | SdkPathParameter | SdkHeaderParameter)` types and use the `serializedName` on the type.